File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/IGameStateManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.IGameState;

public interface IGameStateManager {

    IGameState retrieve();

    void TogglePause();

    void TogglePause(boolean isPaused);

    void SetGameOver();

    void Reset();

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/IGameStateManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.IGameState

interface IGameStateManager {

    fun retrieve(): IGameState

    fun TogglePause()

    fun TogglePause(isPaused: Boolean)

    fun SetGameOver()

    fun Reset()

}
